tests/test_attribute_builder.py

```diff
@@ -95,68 +129,83 @@ def test_two_builders_second_builder():
 
 
 def test_two_builders_bad_args():
-    with pytest.raises(TypeError):
+    with pytest.raises(BuilderNotFoundException):
         TwoBuildersAttr.build([])
 
 
 @irdl_attr_definition
 class BuilderDefaultArgAttr(Data[str]):
     name = "test.builder_default_arg_attr"
-    param: str
 
     @staticmethod
     @builder
-    def from_int(data: int) -> BuilderDefaultArgAttr:
-        return BuilderDefaultArgAttr(str(data))
+    def from_int(data1: int,
+                 data2: int = 42,
+                 data3: int = 17) -> BuilderDefaultArgAttr:
+        return BuilderDefaultArgAttr(f"{data1}, {data2}, {data3}")
 
     @staticmethod
-    def parse_parameter(parser: Parser) -> None:
-        pass
+    def parse_parameter(parser: Parser) -> str:
+        raise NotImplementedError()
 
     @staticmethod
     def print_parameter(data: str, printer: Printer) -> None:
-        pass
+        raise NotImplementedError()
 
 
-def builder_default_arg_default():
+def test_builder_default_args_too_few_args():
+    with pytest.raises(BuilderNotFoundException):
+        BuilderDefaultArgAttr.build()
+
+
+def test_builder_default_args_only_required_args():
     attr = BuilderDefaultArgAttr.build(4)
-    assert attr == BuilderDefaultArgAttr("40")
+    assert attr == BuilderDefaultArgAttr("4, 42, 17")
+
+
+def test_builder_default_args_only_required_args_and_one():
+    attr = BuilderDefaultArgAttr.build(4, 5)
+    assert attr == BuilderDefaultArgAttr("4, 5, 17")
 
 
-def builder_default_arg_arg():
-    attr = BuilderDefaultArgAttr.build(4, 2)
-    assert attr == BuilderDefaultArgAttr("42")
+def test_builder_default_args_all_args():
+    attr = BuilderDefaultArgAttr.build(4, 5, 6)
+    assert attr == BuilderDefaultArgAttr("4, 5, 6")
+
+
+def test_builder_default_args_too_many_args():
```

Comment:
try to drop "too" from the name

Reply:
Done.
I'm actually curious, why is this bad practice?